plugins/outputs/datadog/datadog.go

```diff
@@ -136,17 +122,44 @@ func (d *Datadog) authenticatedUrl() string {
 	return fmt.Sprintf("%s?%s", d.apiUrl, q.Encode())
 }
 
-func buildMetrics(m telegraf.Metric) (map[string]Point, error) {
-	ms := make(map[string]Point)
-	for k, v := range m.Fields() {
-		var p Point
-		if err := p.setValue(v); err != nil {
-			return ms, fmt.Errorf("unable to extract value from Fields, %s", err.Error())
+// Convert a telegraf metric to datadog metrics;
+// we need a separate metric for each field.
+// (also has magic for statsd field names)
+func buildMetrics(m telegraf.Metric) ([]*Metric, error) {
+	var datadogMetrics []*Metric
+	tags := m.Tags()
+	metricType := tags["metric_type"]
```

Comment:
go maps give back the zero of the type if it doesn't exist? Though admittedly I haven't tested this :)